ingestion/src/metadata/ingestion/source/pipeline/nifi/connection.py

```diff
@@ -23,11 +24,19 @@ def get_connection(connection: NifiConnection) -> NifiClient:
     """
     Create connection
     """
+    if isinstance(connection.nifiConfig, UsernamePassword):
+        return NifiClient(
+            host_port=connection.hostPort,
+            username=connection.nifiConfig.username,
+            password=connection.nifiConfig.password.get_secret_value(),
```

Comment:
I think either we make the password field required or we shall handle the none password here otherwise it might break if user does not provide any password!